src/ec/packages/ec-component-media-container/media-container.html.twig

```diff
@@ -0,0 +1,75 @@
+{% spaceless %}
+{#
+  Parameters:
+  - "alt" (string) (default: ''),
+  - "image" (string) (default: ''),
+  - "extra_classes" (string) (default: ''),
+  - "sources" (array) (default: []): format: [
+      {
+        "src" (string) (default: ''),
+        "type" (string) (default: ''),
+      },
+      ...
+    ],
+  - "tracks" (array) (default: []): format: [
+      {
+        "src" (string) (default: ''),
+        "kind" (string) (default: ''),
+        "srcLang" (string) (default: ''),
+        "label" (string) (default: ''),
+      },
+      ...
+    ],
+  - "description" (string) (default: ''),
+  - "extra_attributes" (array) (default: []): format: [
+      {
+        "name" (string) (default: ''),
+        "value" (string) (default: '')
+      },
+      ...
+    ]
+#}
+
+{# Internal properties #}
+
+{% set _css_class = 'ecl-media-container' %}
+{% set _image = image|default('') %}
+{% set _description = description|default('') %}
+{% set _alt = alt|default('') %}
+{% set _extra_attributes = '' %}
+{% set _sources = sources|default({}) %}
+{% set _tracks = tracks|default({}) %}
+
+{# Internal logic - Process properties #}
+
+{% if extra_classes is defined and extra_classes is not empty %}
+  {% set _css_class = _css_class ~ ' ' ~ extra_classes %}
+{% endif %}
+
+{% if extra_attributes is defined and extra_attributes is not empty and extra_attributes is iterable %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %} 
+
+{# Print the result #}
+
+<figure class="{{ _css_class }}"{{ _extra_attributes|raw }}>
+{% if _sources is not empty and sources is iterable %}
+  <video class="ecl-media-container__media" controls poster="{{ _image }}" alt="{{ _alt }}">
+  {% for source in _sources %}
+    <source type="{{ source.type }}" src="{{ source.src }}" />
+  {% endfor %}
+{% if tracks is not empty and tracks is iterable %}
+  {% for track in _tracks %}
+    <track kind="{{ track.kind }}" srclang="{{ track.srcLang }}" label="{{ track.label }}" src="{{ track.src }}" />
+  {% endfor %}
+{% endif %}
```

Comment:
It's tricky closing the if statement and the video tag after it at this place. I would suggest changes in indentation:

```twig
<figure class="{{ _css_class }}"{{ _extra_attributes|raw }}>
  {% if _sources is not empty and sources is iterable %}
    <video
      alt="{{ _alt }}"
      class="ecl-media-container__media"
      controls
      poster="{{ _image }}"
    >
    {% for source in _sources %}
      <source type="{{ source.type }}" src="{{ source.src }}" />
    {% endfor %}
    {% if tracks is not empty and tracks is iterable %}
      {% for track in _tracks %}
        <track
          kind="{{ track.kind }}"
          label="{{ track.label }}"
          src="{{ track.src }}"
          srclang="{{ track.srcLang }}"
        />
      {% endfor %}
    {% endif %}
    </video>
  {% else %}
    <img
      alt="{{ _alt }}"
      class="ecl-media-container__media"
      src="{{ _image }}"
    />
  {% endif %}
  <figcaption class="ecl-media-container__caption">
    {{ _description }}
  </figcaption>
</figure>

I've placed attributes beneath each other to keep the code to under 80 chars ruler if we had one.

PS, not required to accept this, but please indent if/endif statements to be easy to see on first look, thanks.